pybossa/sched.py

```diff
@@ -65,24 +63,28 @@ def new_task(project_id, sched, user_id=None, user_ip=None,
         'depth_first_all': get_depth_first_all_task
     }
     scheduler = sched_map.get(sched, sched_map['default'])
-
     project = project_repo.get(project_id)
-    # This is here for testing. It removes the random variable to make testing deterministic.
     disable_gold = not project.info.get('enable_gold', True)
```

Comment:
@markroseberry is this needed only for the unit tests?  Or do you use this in QA for testing also?

Reply:
With the changes, the quiz mode unit tests pass without needing this option.  Perhaps we can remove it?

Reply:
Do the tests pass deterministically or probabilistically? I'll look into it. I think I put it there for testing quiz mode vs. non-quiz mode, the test being you get only gold in quiz mode and only non-gold in non-quiz mode.

Reply:
This test **test_passed_quiz_normal_task** from **test_quiz_mode.py** fails. What exactly you are testing on it? Why we need to confirm that the user gets a normal task after a quiz is passed?

Reply:
@jayflo, I changed the gold task probability to 1/2 and verified that the test _DOES_ fail intermittently about half the time. So we need to keep that setting and add it back to the unit test. I'll create a PR.

Reply:
@mariarivera30, we need to test that user gets the normal mixture of scheduled tasks after a quiz is passed. Since I modified the scheduler code to only give gold tasks for quiz mode, I wanted to make sure that I didn't break the scheduler behavior in non-quiz mode. So it's sort of like a regression test. 

Now since the normal behavior of the scheduler is probabilistic, that is very hard to test. So I added the setting to only schedule non-gold tasks in non-quiz mode to make it testable.